manage/upgrading/version_specific_migration/upgrade_to_python3.rst

```diff
@@ -209,74 +217,77 @@ Precompile will be run every time you run buildout. If you want to avoid running
 ---------------------
 
 As a next step we recommend that you try to start the instance with your add-on.
-This will fail on all import errors (e.g. relative imports that are not allowed in Python 3).
-If it works you can try to install the add-on.
+This will fail on all import errors (e.g., relative imports that are not allowed in Python 3).
+If it works then you can try to install the add-on.
 
 You need to fix all issues that appear before you can do manual testing to check for big, obvious issues.
 
 
 Common Issues during startup
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The following issues will abort your startup and you need to fix them before you are able to test the functionality by hand or run tests.
+The following issues will abort your startup.
+You need to fix them before you are able to test the functionality by hand or run tests.
 
 
 A - Class Advice
 ^^^^^^^^^^^^^^^^
 
-This kind of error message
+This kind of error message:
 
 .. code-block:: shell
 
-   TypeError: Class advice impossible in Python3.  Use the @implementer class decorator instead.
+    TypeError: Class advice impossible in Python3.  Use the @implementer class decorator instead.
 
 tells you that there is a class that is using an ``implements`` statement which needs to be replaced by the ``@implementer`` decorator.
 
-Example, this kind of code:
+For example, this kind of code:
 
 .. code-block:: python
 
-   from zope.interface import implements
+    from zope.interface import implements
 
-   class Group(form.BaseForm):
-       implements(interface.IGroup)
-       …
+    class Group(form.BaseForm):
+        implements(interface.IGroup)
 
 needs to be replaced with:
 
 .. code-block:: python
 
-   from zope.interface import implementer
+    from zope.interface import implementer
 
-   @implementer(interfaces.IGroup)
-   class Group(form.BaseForm):
-       …
+    @implementer(interfaces.IGroup)
+    class Group(form.BaseForm):
 
-The same is the case for `provides(IFoo)` and some other Class advices. These need to be replaced with their respective decorators like `@provider`.
+The same is the case for ``provides(IFoo)`` and some other Class advices.
+These need to be replaced with their respective decorators like ``@provider``.
 
 
 B - Relative Imports
 ^^^^^^^^^^^^^^^^^^^^
 
-Relative imports like `import permissions` are no longer permitted. Use `from collective.package import permissions` or `from . import permissions` (not recommended)
+Relative imports like ``import permissions`` are no longer permitted.
+Use ``from collective.package import permissions`` or ``from . import permissions`` (not recommended).
```

Comment:
Why did we mention a not recommended way?

Reply:
Agreed, it should be removed, but I'm not 100% sure whether only the latter of the two or both import styles should be removed. It's not clear.

Reply:
@plone/framework-team can you shed some light here? Thanks!

Reply:
We prefer to not use the dot b/c the full path increases readability. Not always you exactly know where you're standing while reading code, like after a search or at github.

Reply:
Fixed in 8278c92.